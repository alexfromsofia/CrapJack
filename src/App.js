import React from 'react';
import { StoreProvider } from './store/StoreProvider';
import { initialState, rootReducer, init } from './store';
import { StateInspector } from 'reinspect';
import Game from './components/Game';

function App() {
  return (
    <StateInspector name="Crap Jack Game">
      <StoreProvider
        initialState={initialState}
        reducer={rootReducer}
        init={init}
      >
        <div className="wrapper">
          <Game />
        </div>
      </StoreProvider>
    </StateInspector>
  );
}

export default App;
