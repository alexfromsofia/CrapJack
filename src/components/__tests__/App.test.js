import React from 'react';
import { shallow } from 'enzyme';
import App from '../../App';
it('renders without crashing', () => {
  shallow(<App />);
});

it('Matches Snapshot', () => {
  const wrapper = shallow(<App />);

  expect(wrapper).toMatchSnapshot();
});
