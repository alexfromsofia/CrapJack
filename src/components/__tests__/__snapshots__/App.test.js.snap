// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<StateInspector
  name="Crap Jack Game"
>
  <StoreProvider
    init={[Function]}
    initialState={
      Object {
        "dealer": Object {
          "cards": Array [],
          "isWinner": false,
          "name": "Dealer",
          "score": 0,
          "wins": 0,
        },
        "deckId": null,
        "gameId": 0,
        "isCleaned": true,
        "isPlaying": false,
        "isRevealed": false,
        "player": Object {
          "cards": Array [],
          "isWinner": false,
          "name": "You",
          "score": 0,
          "wins": 0,
        },
        "shouldDisplayRules": false,
        "systemMessage": "Welcome to Crap Jack!",
      }
    }
    reducer={[Function]}
  >
    <div
      className="wrapper"
    >
      <Game />
    </div>
  </StoreProvider>
</StateInspector>
`;
